### DISCUSSION TITLE
I write my cli app, how to add native support for nushell table output?

### DISCUSSION THREAD

**Discussion by markfswe** [nushell/nushell#10938] [Q&A] [answered]
I could not find any articles in book / cookbook like "How to add support for nushell in your software".

For example I have a nodejs CLI which can output tables using ASCII symbols. I would like to add check if user's shell is nu so I would use it (API?) to print nice and fancy tables.

Maybe I am missing something obvious. I have some of variants in my head, but I am not sure yet.

**Comment by fdncred** [comment]
In order to make nushell tables you'll have to have nushell Values and call the table command. I don't think anything like this is documented anywhere but you can have a look at how couchbase shell integrated nushell https://github.com/couchbaselabs/couchbase-shell. Sounds like they did pretty much what you're wanting to do except they took it further and added a bunch of commands.

**Reply by markfswe** [reply]
Idk, it's a shell which calls other shell and uses same language, uses nu rust API. I could investigate it and look through the code, but I don't think this is what am I looking for

**Reply by fdncred** [reply, marked as answer]
Oh, I thought you were writing a rust CLI and wanted to integrate nushell into it? My mistake. The only way that I can think of it that's easy to use is to make your nodejs CLI output json. Then you can use `<your app> | from json` or `<your app> | from json --objects`.

There's really only a few ways to do this.
1. integrate nushell, which you don't want to do
2. output json, which I just now suggested
3. write a nushell custom command to parse the output
4. ?

**Reply by markfswe** [reply]
Sure. I think i'll just invoke user's nu binary and pass data to it.

**Reply by fdncred** [reply]
yup, you can do that too.

**Reply by gcarrarom** [reply]
That's an interesting idea @ZerdoX-x , did you end up implementing it? I'm thinking of doing the same for some of my CLIs.
In my head it is a bit odd to allow execution of a binary in your CLI though. Something feels off from doing so; still much better than having to create an alias/module/script in my nushell environment to get proper parsing.
